core/arch/arm/mm/core_mmu.c

```diff
@@ -967,7 +967,9 @@ static void check_pa_matches_va(void *va, paddr_t pa)
 			panic("issue in linear address space");
 		return;
 	}
-	if (v >= (CFG_TEE_LOAD_ADDR & ~CORE_MMU_PGDIR_MASK) &&
+	/* wait pager is initialize before looking into pager data */
+	if (cpu_mmu_enabled() &&
```

Comment:
regarding the `inline` to be removed:
I would rather add this `get_linear_map_end()` function in the tee_pager.h so that it remains pager scope. `tee_pager_get_linear_memory_end()` ?

Sorry i have pushed your patch to the wrong branch. it ended in another pull :( 
i will push back.